src/Source/DoctrineODMQuerySourceIterator.php

```diff
@@ -127,30 +161,38 @@ public function rewind()
      */
     public function setDateTimeFormat($dateTimeFormat)
     {
-        $this->dateTimeFormat = $dateTimeFormat;
-    }
+        @trigger_error('Passing a dateTimeFormat string is deprecated since 1.7.2, to be removed in 2.0. '.
+            'The source should be initialized in the constructor with an array of DataFormatterInterfaces instead',
+            E_USER_DEPRECATED);
 
-    /**
-     * @return string
-     */
-    public function getDateTimeFormat()
-    {
-        return $this->dateTimeFormat;
+        $newFormatter = new DateTimeFormatter($dateTimeFormat);
+        $this->formatters[$newFormatter->getPriority()] = $newFormatter;
     }
 
     /**
-     * @param $value
+     * @param mixed        $target
+     * @param PropertyPath $propertyPath
      *
-     * @return null|string
+     * @return mixed
```

Comment:
mixed what? Please list possible types separated with `|` (keep the `mixed` though)